pandas/core/arrays/period.py

```diff
@@ -778,7 +784,7 @@ def _add_delta_tdi(self, other):
         assert isinstance(self.freq, Tick)  # checked by calling function
 
         delta = self._check_timedeltalike_freq_compat(other)
-        return self._addsub_int_array(delta, operator.add).asi8
```

Comment:
huh? what are you adding a new kwarg for?

Reply:
We still call `_addsub_int_array` internally from a few places (here and in DateOffset `apply_index` methods) and want to suppress the warning.

But actually now that the warning is a one-liner, it is less cumbersome to put it directly in `__add__/__sub__`.  Will change.